src/main/java/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionService.java

```diff
@@ -64,7 +62,11 @@ public TranslatedNonMatchingResponseBody translateSuccessResponse(List<Assertion
         validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance);
 
         String nameID = mdsAssertion.getSubject().getNameID().getValue();
+
         LevelOfAssurance levelOfAssurance = extractLevelOfAssuranceFrom(authnAssertion);
+        LevelOfAssuranceValidator levelOfAssuranceValidator = new LevelOfAssuranceValidator();
```

Comment:
In general we don't `new` our validators in the class as it makes it harder to unit test - dependencies should come in via the constructor (and thus the factory)

Reply:
Done👍🏼